app/src/main/java/tech/ula/ui/AppListFragment.kt

```diff
@@ -120,7 +130,7 @@ class AppListFragment : Fragment() {
 
             val preferredServiceType = appListViewModel.getAppServiceTypePreference(selectedApp)
             if (preferredServiceType.isEmpty()) {
-                selectServiceTypePreference(selectedApp)
+                getCredentialsAndStart(selectedApp = selectedApp)
             } else {
                 val serviceIntent = Intent(activityContext, ServerService::class.java)
```

Comment:
The service will now start without password and username information if the preferences have been filled. If the session didn't successfully start and reach the end of `startSession`, the changes to the session won't have been successfully updated, meaning the username and password for the session will be incorrect moving forward indefinitely.